src/SIM/SIMbase.h

```diff
@@ -760,9 +760,11 @@ class SIMbase : public SIMinput, public SIMdependency
   //! \param[in] nodes Total number of global nodes in model
   //! \param[in] basis The basis to specify number of DOFs for
   //! \param[in] nndof Number of DOFs on given basis
-  std::vector<int> setupAdditionalMADOF(const PatchVec& myModel, size_t nodes,
-                                        unsigned char basis,
-                                        unsigned char nndof) const;
+  //! \param[out] madof Generated MADOF array
+  void setupAdditionalMADOF(const PatchVec& myModel, size_t nodes,
+                            unsigned char basis,
+                            unsigned char nndof,
+                            std::vector<int>& madof) const;
 
```

Comment:
Siden setupAdditionalMADOF er private er det vel overflødig å sende inn myModel og nodes=this->getNoNodes(true) som argumenter, når man like godt kan aksessere disse direkte fra innsiden av metoden?

Dessuten, vet ikke hvor mange forskjellige kombinasjoner av basis og nndof som måtte forekomme i en modell, men men kunne vel også tenke seg å beregne alle de nødvendige MADOF-ene i forkant (f.eks. i preprocess), å slippe å sjekke om det er gjort i hvert [inject|extract]PatchSolution kall. Bare en tanke altså, trenger ikke å gjøre om på dette (som sikkert funker) nå.